src/eos-crash-metrics.c

```diff
@@ -0,0 +1,183 @@
+              const char *ostree_url)
+{
+  g_autoptr(GHashTable) ht = g_hash_table_new_full (g_str_hash,
+                                                    g_str_equal,
+                                                    g_free,
+                                                    (GDestroyNotify) g_variant_unref);
+
+  g_hash_table_insert (ht, g_strdup ("binary"), g_variant_new_string (binary));
+  g_hash_table_insert (ht, g_strdup ("signal"), g_variant_new_int16 (signal));
+  g_hash_table_insert (ht, g_strdup ("timestamp"), g_variant_new_int16 (timestamp));
+  g_hash_table_insert (ht, g_strdup ("ostree_commit"), g_variant_new_string (ostree_commit != NULL ? ostree_commit : ""));
+  g_hash_table_insert (ht, g_strdup ("ostree_commit"), g_variant_new_string (ostree_url != NULL ? ostree_url : ""));
+
+  g_message("%s %s", ostree_commit, ostree_url);
+
+  emtr_event_recorder_record_event_sync (emtr_event_recorder_get_default (),
+                                         PROGRAM_DUMPED_CORE_EVENT,
+                                         hashtable_to_variant (ht));
+}
+
+static OstreeSysroot *
+load_ostree_sysroot (void)
+{
+  OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  g_autoptr(GError) error = NULL;
+
+  if (!ostree_sysroot_load (sysroot, NULL, &error))
+    {
+      g_warning ("Unable to get current OSTree sysroot: %s", error->message);
+      return NULL;
+    }
+
+  return sysroot;
+}
+
+static char *
+get_eos_ostree_repo_url (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  char *url = NULL;
+
+  config = ostree_repo_get_config (repo);
+
+  if (!(url = g_key_file_get_string (config, "remote \"eos\"", "url", &error)))
+    {
+      g_warning ("Unable to read OSTree config for eos remote URL: %s", error->message);
+      return NULL;
+    }
+
+  return url;
+}
+
+static char *
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *refspec = NULL;
+  g_autofree char *commit = NULL;
+
+  if (!deployment)
+    {
+      g_warning ("OSTree deployment is not currently booted, cannot read state");
+      return NULL;
+    }
+
+  config = ostree_deployment_get_origin (deployment);
+
+  if (!ostree_sysroot_get_repo (sysroot, &repo, NULL, &error))
+    {
+      g_warning ("Unable to read repo in sysroot: %s", error->message);
+      return NULL;
+    }
+
+  if (!(refspec = g_key_file_get_string (config, "origin", "refspec", &error)))
+    {
+      g_warning ("Unable to read OSTree refspec for booted deployment: %s", error->message);
+      return NULL;
+    }
+
+  if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &commit, &error))
+    {
+      g_warning ("Unable to resolve revision for refpsec %s: %s", refspec, error->message);
+      return NULL;
+    }
+
+  return commit;
```

Comment:
You need to return `g_steal_pointer (&commit)`, otherwise `commit` will be freed when leaving the scope

Reply:
Decided to just make it not-autofree (since we only allocate it at the very end anyway)